examples/mysql-and-postgres/main.tf

```diff
@@ -87,29 +97,84 @@ module "postgresql-db" {
   source           = "../../modules/postgresql"
   name             = "example-postgresql-${random_id.name.hex}"
   database_version = "${var.postgresql_version}"
-  project_id       = "${data.google_client_config.current.project}"
+  project_id       = "${var.project_id}"
   zone             = "c"
 
   ip_configuration = [{
+    ipv4_enabled = true
+
     authorized_networks = [{
       name  = "${var.network_name}"
       value = "${google_compute_subnetwork.default.ip_cidr_range}"
     }]
   }]
 }
 
+# We define a VPC peering subnet that will be peered with the
+# Cloud SQL instance network. The Cloud SQL instance will
+# have a private IP within the provided range.
+resource "google_compute_global_address" "google-managed-services-default" {
+  provider      = "google-beta"
+  project       = "${var.project_id}"
+  name          = "private-ip-address"
+  purpose       = "VPC_PEERING"
+  address_type  = "INTERNAL"
+  prefix_length = 16
+  network       = "${google_compute_network.default.self_link}"
+}
+
+# Creates the peering with the producer network.
+resource "google_service_networking_connection" "private_vpc_connection" {
```

Comment:
Should this eventually be included in the module itself?

Reply:
Defined as a submodule, see below

Reply:
@mmontan What is your motivation for creating a new submodule?

Reply:
The peering and assignment of IPs can only be done once for each VPC,  and it will be used by all Cloud SQL instances with private ips in that VPC.

Because multiple Cloud SQL private instances can be added to the same VPC, it seems that it would be cleaner to have it as an independent module that it is created before the instances.
 
Do you have other ideas on how this can be done?